lib/IRGen/LoadableByAddress.cpp

```diff
@@ -4077,9 +4087,15 @@ class RewriteUser : SILInstructionVisitor<RewriteUser> {
              "caseBB has a payload argument");
 
       SILBuilder caseBuilder = assignment.getBuilder(caseBB->begin());
+      SILType eltType =
```

Comment:
I would prefer to keep the type lowering code out of this phase.

It should be possible to read lowered (mapped) types out of the "input" SIL to this phase.

In this case, the result type is the switch_enum's successor basic block argument  type.

This should work, I think:
```
diff --git a/lib/IRGen/LoadableByAddress.cpp b/lib/IRGen/LoadableByAddress.cpp
index d76c9fc3310..2502d55bc70 100644
--- a/lib/IRGen/LoadableByAddress.cpp
+++ b/lib/IRGen/LoadableByAddress.cpp
@@ -4078,7 +4078,8 @@ protected:
 
       SILBuilder caseBuilder = assignment.getBuilder(caseBB->begin());
       auto *caseAddr =
-          caseBuilder.createUncheckedTakeEnumDataAddr(loc, opdAddr, caseDecl);
+          caseBuilder.createUncheckedTakeEnumDataAddr(loc, opdAddr, caseDecl,
+                                                      caseArg->getType().getAddressType());
 
       if (assignment.isLargeLoadableType(caseArg->getType())) {
         assignment.mapValueToAddress(caseArg, caseAddr);
```

Reply:
Oh, indeed here!